### ISSUE TITLE
TODO 36: Test determinism — seed _gdino_outputs/_gdino_targets helpers

### ISSUE THREAD

**Issue by hongthepotato** [hongthepotato/Grounded-Segment-Anything#85]
## Summary

Two helper functions in [`tests/integration/test_ml_pipeline_cpu.py`](https://github.com/hongthepotato/Grounded-Segment-Anything/blob/agentic/tests/integration/test_ml_pipeline_cpu.py#L57-L72) call `torch.rand(...)` without seeding. ~6 existing tests in that file consume those helpers and are technically non-deterministic. Failures involving boundary conditions won't reproduce on rerun.

**Priority:** P2.

## Where

`tests/integration/test_ml_pipeline_cpu.py:57-72`:

```python
def _gdino_outputs(batch=1, num_queries=5, num_tokens=256):
    return {
        "pred_logits": torch.rand(batch, num_queries, num_tokens),  # unseeded
        "pred_boxes": torch.rand(batch, num_queries, 4),             # unseeded
    }

def _gdino_targets(batch=1, num_gt=2, num_tokens=256):
    return [{
        "labels": torch.zeros(num_gt, dtype=torch.long),
        "boxes": torch.rand(num_gt, 4),                              # unseeded
        "token_labels": torch.zeros(num_gt, num_tokens),
    } for _ in range(batch)]
```

**Tests that consume these helpers:**
- `TestBuildCriterionBasics::test_returns_non_empty_dict`
- `TestBuildCriterionBasics::test_main_loss_keys_present`
- `TestBuildCriterionBasics::test_gradient_flows_through_losses`
- `TestBuildCriterionBasics::test_multi_class_produces_finite_losses`
- `TestBuildCriterionBounds::*` (4 tests)
- `TestBuildCriterionPerfectDetection::test_near_zero_ce_on_low_confidence_zero_token_labels`

## Why

- Tests are technically non-deterministic.
- Failures involving boundary conditions (e.g. `class_error` exactly at 0 or 100, GIoU near ±1) won't reproduce on rerun.
- No flake observed yet, but the surface is real.

## Fix (pick one)

```python
# Option A — autouse fixture at module top
@pytest.fixture(autouse=True)
def _seed_rng():
    torch.manual_seed(0)
```

```python
# Option B — pass generator into rand calls
def _gdino_outputs(batch=1, num_queries=5, num_tokens=256, seed=0):
    g = torch.Generator().manual_seed(seed)
    return {
        "pred_logits": torch.rand(batch, num_queries, num_tokens, generator=g),
        "pred_boxes": torch.rand(batch, num_queries, 4, generator=g),
    }
```

```python
# Option C — seed at the top of each helper (simplest)
def _gdino_outputs(batch=1, num_queries=5, num_tokens=256):
    torch.manual_seed(0)
    return {...}
```

**Recommendation:** Option A. Two lines, applies to all consumers, no helper-signature churn.

## Acceptance criteria

- [ ] All 6 affected tests reproduce identical loss values on repeated runs
- [ ] No assertion changes — same tests, just deterministic now
- [ ] Full suite still 2348 passing

## Surfaced via

`/review` (testing specialist) on branch `integration-tests/ml-pipeline-cpu` (2026-05-08).
